packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -28,4 +47,53 @@ export class Snapshot {
   async delete() {
     // TODO: CDX-359: Delete snapshot once previewed
   }
+
+  get lastestReport(): ResourceSnapshotsReportModel {
+    if (this.model.reports === undefined) {
+      throw new Error(
+        `No detailed report found for the snapshot ${this.snapshotId}`
+      );
+    }
+    return this.model.reports.slice(-1)[0];
+  }
+
+  private get snapshotId() {
+    return this.model.id;
+  }
+
+  private async refreshSnapshotData() {
+    this.model = await (this.client.get as any)(this.model.id, {
+      includeReports: false,
+    });
+  }
+
+  private async isIdle() {
+    await this.refreshSnapshotData();
+    const lastReport = this.model.reports?.slice(-1)[0];
```

Comment:
That seems very similar to `get lastestReport()` ? 🤔

Reply:
yeah, that was a bad commit. I replaced it with the get latestReport :P